src/algebra/zset/mod.rs

```diff
@@ -47,10 +47,57 @@ pub trait IndexedZSet: Batch<Time = ()> + GroupValue + NumEntries {
     {
         self.distinct()
     }
+
+    /// Returns an iterator over updates in the indexed Z-set.
+    fn iter(&self) -> IndexedZSetIterator<Self> {
+        IndexedZSetIterator::new(self.cursor())
+    }
 }
 
 impl<Z> IndexedZSet for Z where Z: Batch<Time = ()> + GroupValue + NumEntries {}
 
+/// Iterator over `(key, value, weight)` tuples of an indexed Z-set.
+pub struct IndexedZSetIterator<'a, Z>
+where
+    Z: IndexedZSet,
+{
+    cursor: Z::Cursor<'a>,
+}
+
+impl<'a, Z> IndexedZSetIterator<'a, Z>
+where
+    Z: IndexedZSet,
+{
+    /// Returns an iterator of `(key, value, weight)` over the items that
+    /// `cursor` visits.
+    fn new(cursor: Z::Cursor<'a>) -> Self {
+        Self { cursor }
+    }
+}
+
+impl<'a, Z> Iterator for IndexedZSetIterator<'a, Z>
+where
+    Z: IndexedZSet,
+{
+    type Item = (Z::Key, Z::Val, Z::R);
```

Comment:
Actually, couldn't we just use the lifetime here?
```suggestion
    type Item = (&'a Z::Key, &'a Z::Val, Z::R);
```

Reply:
I tried hard to make it work. Don't think it's possible.